pallets/multisig/src/lib.rs

```diff
@@ -544,6 +546,11 @@ decl_module! {
             let perms = Permissions::from_pallet_permissions(
                 iter::once(PalletPermissions::entire_pallet(NAME.into()))
             );
+
+            // Ensure we won't have too many keys.
+            let record = <DidRecords<T>>::get(did);
+            ensure_length_ok::<T>(record.secondary_keys.len().saturating_add(1))?;
+
```

Comment:
I would move this check up by the `ensure_key_did_unlinked`.  Doing the checks as early as possible helps to decrease the cost to the node for bad requests.  Here it is not saving much, but would be nice anyways.